src/main/java/com/amazon/opendistroforelasticsearch/jobscheduler/scheduler/JobScheduler.java

```diff
@@ -158,7 +162,20 @@ boolean reschedule(ScheduledJobParameter jobParameter, JobSchedulingInfo jobInfo
             JobExecutionContext context = new JobExecutionContext();
             context.setExpectedExecutionTime(jobInfo.getExpectedPreviousExecutionTime());
             context.setJobVersion(version);
-            jobRunner.runJob(jobParameter, context);
+
+            if (jobParameter.getLockDurationSeconds() == null) {
+                jobRunner.runJob(jobParameter, context);
+            } else {
+                final LockModel lock =
+                    lockService.acquireLock(jobInfo.getIndexName(), jobInfo.getJobId(), jobParameter.getLockDurationSeconds());
+                if (lock != null) {
+                    log.info("We have acquired lock. Running job.");
+                    jobRunner.runJob(jobParameter, context);
```

Comment:
Do you think a try catch around runJob would be useful so you can log in the catch that an error occurred and we now have to wait for the lock expiration time?

Reply:
As long as we long when runJob fails, I don't think we need to log specifically about waiting for  lock to expire. As users who use the locking option should already know that they will need to wait.